crates/hir-ty/src/display.rs

```diff
@@ -1775,21 +1775,34 @@ fn write_bounds_like_dyn_trait(
     Ok(())
 }
 
+#[derive(Clone, Copy)]
+pub enum TraitRefFormat {
+    SelfAsTrait,
+    SelfImplementsTrait,
+    OnlyTrait,
+}
```

Comment:
I think it would make more sense to implement HirDisplay directly for `hir::TraitRef` and have that by default omit the self type. Then we don't need to expose this enum beyond the `hir` crate nor do we need the formatting wrapper struct

Reply:
That makes sense!  Certainly much cleaner, I've made that change.  Thanks for identifying it